conformance/lang/expressions/unary-expression/unary_complement_expression.balt

```diff
@@ -0,0 +1,523 @@
+
+    int:Signed32? _ = ~d; // @error static type of unary-complement with operands of optional int subtypes is int optional
+
+    int:Unsigned8? _ = ~e; // @error static type of unary-complement with operands of optional int subtypes is int optional
+
+    int:Unsigned16? _ = ~f; // @error static type of unary-complement with operands of optional int subtypes is int optional
+
+    int:Unsigned32? _ = ~g; // @error static type of unary-complement with operands of optional int subtypes is int optional
+
+    Ints? _ = ~h; // @error static type of unary-complement with operands of optional int subtypes is int optional
+}
+
+Test-Case: output
+Description: Test unary complement operator with unary-plus and unary-minus operators.
+Fail-Issue: ballerina-lang#32702
+Labels: DecimalNumber, function-call-expr, unary-complement, unary-minus, unary-plus
+
+function init() {
+    io:println(+~1); // @output -2
+    io:println(~+1); // @output -2
+    io:println(~-1); // @output 0
+    io:println(-~1); // @output 2
+
+    io:println(minusTwo(42)); // @output 40
+    io:println(minusTwo(-98)); // @output -100
+
+    io:println(addTwo(42)); // @output 44
+    io:println(addTwo(-98)); // @output -96
+}
+
+public function minusTwo(int i) returns int {
+    return ~-~-i;
+}
+public function addTwo(int i) returns int {
+    return -~-~i;
+}
+
+Test-Case: output
+Description: Test unary complement with field access expression.
+Labels: int, field-access-expr, record-type, unary-complement
+
+type Record record {
+    int x;
+};
+
+function init() {
+    Record b = {x: 10};
+    int c = ~b.x;
+    io:println(c); // @output -11
+}
+
+Test-Case: output
+Description: Test unary complement with member access expression.
+Labels: array-type, int, list-constructor, member-access-expr, unary-complement
+
+function init() {
+    int[] d = [30, 40];
+    int e = ~d[0] ;
+    io:println(e); // @output -31
+}
+
+Test-Case: output
+Description: Test unary complement with a function call.
+Labels: DecimalNumber, function-call-expr, int, module-class-defn, new-expr, unary-complement
+
+class Class {
+    function getInt() returns int {
+        return 60;
+    }
+}
+
+function init() {
+    Class f = new;
+    int g = ~f.getInt();
+    io:println(g); // @output -61
+}
+
+Test-Case: output
+Description: Test unary complement with let expression.
+Labels: DecimalNumber, int, let-expr, unary-complement
+
+function init() {
+    int h = let int h1 = ~3 in h1;
+    io:println(h); // @output -4
+}
+
+Test-Case: output
+Description: Test unary complement with type cast.
+Labels: DecimalNumber, DecimalFloatingPointNumber, int, type-cast-expr, unary-complement
+
+function init() {
+    int i = ~<int>6;
+    io:println(i); // @output -7
+
+    int j = ~<int>10.5;
+    io:println(j); // @output -11
+}
+
+Test-Case: output
+Description: Test unary complement with typeof.
```

Comment:
Other tests are for when the particular expression kind is the operand expression kind, right? Here it is the other way around. I guess it's OK to have these, but shall we group them separately?